proxy/src/main/java/com/wavefront/agent/ProxyConfig.java

```diff
@@ -1422,4 +1453,37 @@ public int compareTo(@NotNull Object o) {
       return Integer.compare(this.order, other.order);
     }
   }
+
+  /**
+   * Helper function to construct tenant info {@link TenantInfo} object based on input parameters.
+   *
+   * @param serverToServiceClientId the CSP OAuth app id.
+   * @param serverToServiceClientSecret the CSP OAuth app secret.
+   * @param cspOrgId the CSP organisation id.
+   * @param cspAPIToken the CSP API token.
+   * @param token the Wavefront API token.
+   * @param server the server url.
+   * @return constructed tenant info {@link TenantInfo} object.
+   */
+  private TenantInfo constructTenantInfoObject(
+      @Nullable final String serverToServiceClientId,
+      @Nullable final String serverToServiceClientSecret,
+      @Nullable final String cspOrgId,
+      @Nullable final String cspAPIToken,
+      @Nonnull final String token,
+      @Nonnull final String server) {
+
+    TenantInfo tenantInfo;
+    if (StringUtils.isNotBlank(serverToServiceClientId)
+        && StringUtils.isNotBlank(serverToServiceClientSecret)
+        && StringUtils.isNotBlank(cspOrgId)) {
+      tenantInfo =
+          new TenantInfo(serverToServiceClientId, serverToServiceClientSecret, cspOrgId, server);
+    } else if (StringUtils.isNotBlank(cspAPIToken)) {
+      tenantInfo = new TenantInfo(cspAPIToken, server, CSP_API_TOKEN);
```

Comment:
we should log, or crash, if 4 values are set.
Also, add a log info message indicating which method are we using.

Reply:
> we should log, or crash, if 4 values are set. Also, add a log info message indicating which method are we using.

I'll add logs. If 4 values are set we decided to give priority to the clientId, clientSecret